data/tabular/train_test_split.py

```diff
@@ -0,0 +1,158 @@
+import tdc
+import pandas as pd
+import numpy as np
+from typing import List, Tuple, Dict, Union, Optional
+import sys
+from tqdm import tqdm
+from random import Random
+from collections import defaultdict
+from rdkit import Chem
+from rdkit.Chem.Scaffolds import MurckoScaffold
+from rdkit import RDLogger
+from glob import glob
+
+RDLogger.DisableLog("rdApp.*")
+
+
+REPRESENTATIONS = [
+    'SMILES',
+    'SELFIES',
+    'PSMILES',
+    'DeepSMILES',
+    'canonical SMILES',
+    'InChI',
+    'TUCAN',
+    'IUPAC name'
+    ]
+
+
+def print_sys(s):
+    """system print
+
+    Args:
+        s (str): the string to print
+    """
+    print(s, flush=True, file=sys.stderr)
+
+
+def create_scaffold_split(df, seed, frac, entity):
+    """create scaffold split. it first generates molecular scaffold for each molecule and then split based on scaffolds
+    reference: https://github.com/chemprop/chemprop/blob/master/chemprop/data/scaffold.py
```

Comment:
```suggestion
   adapted from https://github.com/chemprop/chemprop/blob/master/chemprop/data/scaffold.py
```

if one-to-one copy, say "copied from". Thanks for adding the reference!